### ISSUE TITLE
[Feature request] List view?

### ISSUE THREAD

**Issue by hipunk** [tborychowski/newtab-bookmarks#9]
Hi,

First of all: Excellent add-on. Why firefox still manages bookmarks in a separate window and not in a tab is beyond me. You fill that gap perfectly. Big bonus points for understanding 'Bookmarks Menu' as a folder name, that way I can see all of my bookmarks without having to rearrange anything. I would suggest making that the default?

Now on the the feature request:
What about a list view? I would like that over the icon view.

Thanks for having created this must-have add-on.

Greetz,

hipunk

**Comment by tborychowski** [comment]
Hi,

Thanks for using it :-)
List-view shouldn't be too difficult.
Leave it with me :-)

**Comment by hipunk** [comment]
Awesome!

**Comment by tborychowski** [comment]
[v3.0](https://addons.mozilla.org/en-US/firefox/addon/newtab-bookmarks/versions/?page=1#version-3.0.0) has just been released.
Enjoy :-)

**Comment by hipunk** [comment]
Awesome! It looks a bit different than I expected it would, but I am satisfied.
Really cool, thanks for listening to your users!
The presets are a very thoughtful addition as well. Very happy right now!

**Comment by tborychowski** [comment]
I'm curious then, what did you have in mind. Would you mind sending a screenshot/mockup or some other example?

**Comment by hipunk** [comment]
Sure!
--warning! wall of text. scroll down to tl;dr if this is too much--
I was only looking for this feature/add-on because I was recently using Opera Browser extensively. Before that I was using Vivaldi for a short period of time. When I went back to good old Firefox, I found I really missed having the bookmarks in a tab. In Firefox they are usually to be found in the star menu, menu bar or an extra window, quite the hassle, in my opinion, as I see tabs as the norm nowadays, not as an exceptional thing.
Here, take a look at what Opera does:
![screen shot 2018-04-17 at 08 51 54](https://user-images.githubusercontent.com/11931747/38853187-ba450904-421c-11e8-9788-589a42e015b2.png)

When hovering over items, the list view characteristics are more obvious:
![screen shot 2018-04-17 at 08 51 59](https://user-images.githubusercontent.com/11931747/38853193-bead9998-421c-11e8-8e19-37f0eb6fc411.png)
Entering folders is not perfect in my opinion, as going to the parent folder is only possible by clicking that link now above the list, but acceptable:
![screen shot 2018-04-17 at 08 52 29](https://user-images.githubusercontent.com/11931747/38853493-b1c30e92-421d-11e8-82dd-37b2317095e2.png)




Here is what Vivaldi looks like:
![screen shot 2018-04-17 at 08 54 23](https://user-images.githubusercontent.com/11931747/38853258-feaceb02-421c-11e8-8b15-5c8537279852.png)
It is a list view in the classical sense:
![screen shot 2018-04-17 at 08 54 30](https://user-images.githubusercontent.com/11931747/38853281-0d0d859e-421d-11e8-82bf-f57d7a1f431e.png)
Entering folders is nice and easy as it can be done using the little triangle on the left:
![screen shot 2018-04-17 at 08 54 34](https://user-images.githubusercontent.com/11931747/38853312-25189b92-421d-11e8-946f-15068faad3de.png)



I was looking more for something in that fashion, but I don't want to complain as I'm thankful for having "at least" your add-on now 😁
I am currently back to the icon view in your add-on (sorry for wasting your time, but I've got too many bookmarks and they seem to lay out a tad better in icon view), but if there was a list view like Opera's, I'd prefer that.
I already searched for a quick fix, but found that a list view is quite "complicated" in HTML/CSS as you have to layout everything manually and cannot simply put in a pre-built list view like in most GUI APIs you find in most IDEs.

To get a more simplistic list view, I could simply export all the bookmarks as a HTML file and set that as the new tab page, but then I'd lose the ability to go in and out of folders, and for some strange reason, I'd need an extension for that as well (new tab override), so I might just go with newtab-bookmarks.
I also found two other extensions called BookmarksInTab and New Tab Bookmark. One of them updated one month ago, the other one today, seems like the race is on! 🤣
No, but really: when I first installed your add-on it was the only one I could find, that's why came here first. Also, I think I've rated you 5 stars already but that review was nowhere to be found a few minutes ago. I rated you 5 stars again.

I guess your add-on (in contast to the other 2) has already all the logic necessary to mimic Vivaldi bookmarks, it's merely the cosmetics that are missing now. You can expand folders dynamically and close them again (just like vivaldi does, it just looks different), the others are just static pages, it seems.
**### tl;dr: Here is a (quick and dirty) mockup of what I would love to see:**
![screen shot 2018-04-17 at 09 10 26](https://user-images.githubusercontent.com/11931747/38854799-efe246c6-4221-11e8-93bb-2b7f35703101.png)
Alternative (only a matter of personal taste):
![screen shot 2018-04-17 at 09 10 26alternative](https://user-images.githubusercontent.com/11931747/38854814-fa7a6910-4221-11e8-8c83-c8d24c783bd5.png)

_favicon & bookmark name + the ability to have folders that can be expanded and collapsed_.
No URLs.
I'm aware, I'm asking for a lot here, but might as well try to share the vision instead of absolutely doing nothing, right?
Hope my thoughts help you in making your add-on better.

Big thanks for listening to users.

Greetz and have a nice day/night,

hipunk

**Comment by tborychowski** [comment]
Firstly - thanks for the 5 star ratings!

Secondly - I'm glad that I asked, as that's very different than what I thought :-)
If you don't need the list-view I might actually remove it, as I don't really like how it looks :-D

I think the main difference is that the examples from Opera & Vivaldi are actually full bookmark managers, rather than a new-tab things. And I'm not even trying to aim at building a bookmark-manager, as the extension APIs are very limited and I don't have access to so many things!
(e.g. bookmarks have favicons stored in the browser - I can't access that! I need to use a service to grab them separately!)

Also - all of the other extensions (including the ones that you mentioned) are more "demanding" when it comes to permissions. Mine only needs the bare minimum - access to your bookmarks, while others require access to all pages that you visit. And this is a very important factor to some of my users (myself included) :-) Mine is also fully open-source (except the favicon grabber 3rd parties) with MIT license, which is also important (for transparency, openness and freedom 😄 ).

There's a way to open a bookmark manager in a tab in firefox - just paste this into the addressbar: `chrome://browser/content/places/places.xul`
Or - even better, as it looks very similar to the examples above: `chrome://browser/content/bookmarks/bookmarksPanel.xul`
Maybe you could use this as a start/home page :-)

**Comment by hipunk** [comment]
I understand completely and thank you for trying. I am terribly sorry for having wasted your time by not communicating sufficiently what I meant. 
A complete bookmark manager is overkill, I agree. A sole bookmarks viewer would be sufficient already.

`chrome://browser/content/places/places.xul` is exactly what I missed and I wasn't aware it existed.
As for the list views looks: I think it's okay, but I can see where you're coming from. The default icon view looks very tidy, symmetric and aesthetic.
I like it better that way as well.
Being able to change the coloring is really neat, by the way.

I'm still very happy that this conversation happened, it gave me useful insight I wouldn't have had otherwise.

Bookmarks sidebar is nice, yes. But that sidebar area is already used up by a vertical tab extension for me as I tend to lose focus with tabs on top. Anyways…

I'm happy after rearranging my bookmarks a little bit (I had to scroll quite a bit when I started using it, that's why I'm here. I thought 'There must be a better way!'). It was time for me to do that anyways.

**Comment by tborychowski** [comment]
:-) great!
And you didn't waste my time!
The list view was mostly just a css change - quick & easy, and I learnt some css grid on the way, which is great!
But most of all - this gives me some insight on how people use the extension (which I don't have otherwise, as there are no analytics or any tracking in the extension 😄 ).

Thanks for the good discussion!

**Comment by neomodern** [comment]
I didn't read the entirety of hipunk's messages but I'm also a previous user of Bookmark Tab and I think the list view in this WebExtension would be improved by following it's design for the List View.

It looks like this: 
![](https://i.imgur.com/1IGffua.png)

It's most similar to chrome://browser/content/bookmarks/bookmarksPanel.xul except for a few features:
- You can have up to 3 columns
- You can selectively choose which folders to show in which columns, and resort the order of them
- You can have the same folder appear in multiple columns

I think it's worth improving the List View to follow this design as it's a lot more intuitive than the current one. Thanks for listening.

**Comment by tborychowski** [comment]
@neomodern 
Thanks for your comment.
As much as i understand your use-case, this is drastically different (in both functionality and structure) from my humble extension and would increase its size (and in consequence - maintenance cost).
And I'm not prepared to commit to that (due to a very limited time I can spare) :-)
The assumption for my addon was always: the smallest possible size and fastest possible speed.
And the reason for creating it was that I switched from Safari and this was the only thing that I was missing :-)
(Safari provides only 1 option: selecting a folder to display. mine already has too many options :-))

Anyway, thanks for taking the time to comment, but unfortunately this request is not something I'm planning to add in the near future :-)
